p2p/peerResend.go

```diff
@@ -0,0 +1,99 @@
+package p2p
+
+import (
+	"crypto/sha1"
+	"sync"
+	"time"
+)
+
+type PeerResend struct {
```

Comment:
Not a big fan of that name, it doesn't describe what this class does, it's actually more generic than this name suggest, there is nothing specific about peer or send in this file.
HashTracker? Deduper? HashCache?
I just realized... it is just a kind of LRU cache!

Reply:
going with "PeerResendFilter" which is a boring name but describes what it does